example-app/resources/views/cities/index.blade.php

```diff
@@ -0,0 +1,191 @@
+                    return;
+                }
+
+                $.ajax({
+                    url: `/api/cities/${id}`,
+                    type: 'PUT',
+                    data: {
+                        name: name,
+                        _token: '{{ csrf_token() }}',
+                    },
+                    success: function(response) {
+                        row.find('td:nth-child(2)').text(response.city.name);
+                    },
+                    error: function(xhr) {
+                        if (xhr.responseJSON && xhr.responseJSON.errors) {
+                            alert(xhr.responseJSON.errors.name[0]);
+                        } else if (xhr.responseJSON && xhr.responseJSON.message) {
+                            alert(xhr.responseJSON.message);
+                        } else {
+                            alert('Unkown error editing the city.');
+                        }
+                    }
+                });
+            });
+
+
+            function loadCities(params) {
+                $.ajax({
+                    url: '{{ route('cities.index') }}',
+                    type: 'GET',
+                    data: params,
+                    success: function(response) {
+                        $('.city-container').html(response);
+                    },
+                    error: function(xhr) {
+                        alert('Error loading cities.');
+                    }
+                });
+            }
+
+            let airlineId = '';
+            let sortType = 'id';
+            let orderAsc = true;
+
+            $('#airline-filter').on('change', function() {
+                airlineId = $(this).val();
+                loadCities({
+                    airline_id: airlineId,
+                    sort: sortType,
+                    order: 'asc'
+                });
+                orderAsc = true;
+            });
+
+            $('.city-container').on('click', '.sort', function() {
+                sortType = $(this).data('sort');
+                if(orderAsc){
+                    var order = 'asc';
+                    orderAsc = false;
+                }else{
+                    var order = 'desc';
+                    orderAsc = true;
+                }
+                loadCities({
+                    sort: sortType,
+                    order: order,
+                    airline_id: airlineId
+                });
+            });
+
+            $('.city-container').on('click', '.pagination a', function(e) {
+                e.preventDefault();
+                const page = $(this).attr('href').split('page=')[1];
+                const params = {
+                    page: page,
+                    airline_id: $('#airline-filter').val(),
+                    sort: $('.sort[data-order]').data('sort'),
+                    order: $('.sort[data-order]').data('order')
+                };
+                loadCities(params);
+            });
```

Comment:
Clever! this is for use the laravel paginator links, but to avoid reloading, right?